iml-services/iml-service-queue/src/service_queue.rs

```diff
@@ -4,85 +4,56 @@
 
 use futures::{future, Stream, StreamExt, TryFutureExt, TryStreamExt};
 use iml_rabbit::{
-    basic_consume, connect_to_queue, connect_to_rabbit, purge_queue, BasicConsumeOptions,
-    Connection, ImlRabbitError,
+    basic_consume, connect_to_queue, purge_queue, BasicConsumeOptions, Channel, ImlRabbitError,
 };
 use iml_wire_types::{Fqdn, PluginMessage};
+use thiserror::Error;
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum ImlServiceQueueError {
-    ImlRabbitError(ImlRabbitError),
-    SerdeJsonError(serde_json::error::Error),
-}
-
-impl std::fmt::Display for ImlServiceQueueError {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        match *self {
-            ImlServiceQueueError::ImlRabbitError(ref err) => write!(f, "{}", err),
-            ImlServiceQueueError::SerdeJsonError(ref err) => write!(f, "{}", err),
-        }
-    }
-}
-
-impl std::error::Error for ImlServiceQueueError {
-    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
-        match *self {
-            ImlServiceQueueError::ImlRabbitError(ref err) => Some(err),
-            ImlServiceQueueError::SerdeJsonError(ref err) => Some(err),
-        }
-    }
-}
-
-impl From<ImlRabbitError> for ImlServiceQueueError {
-    fn from(err: ImlRabbitError) -> Self {
-        ImlServiceQueueError::ImlRabbitError(err)
-    }
-}
-
-impl From<serde_json::error::Error> for ImlServiceQueueError {
-    fn from(err: serde_json::error::Error) -> Self {
-        ImlServiceQueueError::SerdeJsonError(err)
-    }
+    #[error(transparent)]
+    ImlRabbitError(#[from] ImlRabbitError),
+    #[error(transparent)]
+    SerdeJsonError(#[from] serde_json::error::Error),
 }
 
 /// Creates a consumer for an iml-service.
 /// This fn will first purge the queue
 /// and then consume from it.
 ///
 /// This is expected to be called once during startup.
-pub fn consume_service_queue(
-    client: Connection,
-    name: &'static str,
-) -> impl Stream<Item = Result<PluginMessage, ImlServiceQueueError>> {
+pub async fn consume_service_queue<'a>(
```

Comment:
Can the lifetime be elided?

Reply:
Apparently I missed this lint. I'll update

Reply:
Hrrm, actually it looks like this lint is faulty. 

Since both `ch` and `name` are borrowed, there is no way to tie the stream to an anonymous lifetime as it can't tell which item it's referring to.